### ISSUE TITLE
DeepSpeed breaks learning rate scheduler checkpointing

### ISSUE THREAD

**Issue by gahdritz** [PyTorchLightning/pytorch-lightning#12576]
## 🐛 Bug

Activating DeepSpeed prevents learning rate schedulers configured in `configure_optimizers` from being loaded properly from checkpoints. I've confirmed using my own setup that activating/deactivating DeepSpeed fixes/reintroduces the issue:

![image](https://user-images.githubusercontent.com/26167235/161350491-3eee7120-b502-4dab-a0dc-c9834b3426af.png)

I'm using a simple learning rate scheduler that warms up the learning rate for one epoch and then flatlines. I ran the model for a couple of epochs (red) and then saved a checkpoint. I loaded the checkpoint with DeepSpeed enabled (green) and the learning rate scheduler was improperly reset. I then disabled DeepSpeed and continued training from the same checkpoint (blue) and the learning rate scheduler resumed as expected.

###  Environment
* CUDA:
        - GPU:
        - available:         False
        - version:           11.3
* Packages:
        - numpy:             1.21.1
        - pyTorch_debug:     False
        - pyTorch_version:   1.10.1+cu113
        - pytorch-lightning: 1.5.10
        - tqdm:              4.62.2
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                -
        - processor:         x86_64
        - python:            3.7.0
        - version:           #1 SMP Wed Dec 1 05:10:33 EST 2021

### Additional context

I tried testing this using version 1.6, but that crashed during the import. In any case, I can't find an issue addressing this bug, so I assume that it's still present.

**Comment by carmocca** [comment]
@gahdritz is this still an issue? Can you provide a reproducible script? You can adapt our [bug report model](https://github.com/Lightning-AI/lightning/blob/master/examples/pl_bug_report/bug_report_model.py)

**Comment by gahdritz** [comment]
It's still an issue, but I'm no longer sure that it's a bug. If I activate DeepSpeed and use a DeepSpeed optimizer, what should I expect to happen with learning rate schedulers configured in `configure_optimizers`? How do those interact with DeepSpeed's own learning rate schedulers?

**Comment by carmocca** [comment]
If the `optimizer` is defined in the config, we use it as is and ignore what's inside `configure_optimizers`. This is the relevant piece of code: https://github.com/Lightning-AI/lightning/blob/07b39c257bac44d82a5d704d79ef103c829597a2/src/pytorch_lightning/strategies/deepspeed.py#L509-L518

We recommend creating them using `configure_optimizers` instead.
